src/AutoRest.CSharp/Mgmt/Output/ResourceContainer.cs

```diff
@@ -31,11 +34,34 @@ public ResourceContainer(OperationGroup operationGroup, BuildContext<MgmtOutputL
         protected override string CreateDescription(OperationGroup operationGroup, string clientPrefix)
         {
             StringBuilder summary = new StringBuilder();
+            string ParentResourceName = GetParentResourceName();
             return string.IsNullOrWhiteSpace(operationGroup.Language.Default.Description) ?
-                $"A class representing collection of {clientPrefix} and their operations over a [ParentResource]. " :
+                $"A class representing collection of {clientPrefix} and their operations over a {ParentResourceName}." :
                 BuilderHelpers.EscapeXmlDescription(operationGroup.Language.Default.Description);
         }
 
+        private string GetParentResourceName()
+        {
+            var parentResourceType = OperationGroup.ParentResourceType(_context.Configuration.MgmtConfiguration);
+
+            switch (parentResourceType)
+            {
+                case ResourceTypeBuilder.ResourceGroups:
+                    return ResourceGroup;
+                case ResourceTypeBuilder.Subscriptions:
+                    return Subscription;
+                case ResourceTypeBuilder.Tenant:
+                    return Tenant;
+                default:
+                    OperationGroup? opGroup = FindParentOperationGroup(parentResourceType);
+                    if (opGroup != null)
+                    {
+                        return opGroup.Resource(Context.Configuration.MgmtConfiguration);
+                    }
+                    return "";
```

Comment:
lets return some default string like "Parent" otherwise the comment sentence won't make any sense.
```suggestion
                    return "Parent";
```